demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py

```diff
@@ -0,0 +1,250 @@
+from demisto_sdk.commands.update_xsoar_config_file.update_xsoar_config_file import \
+    XSOARConfigFileUpdater
+
+UNIT_TEST_DATA = (src_root() / 'commands' / 'update_xsoar_config_file' / 'tests' / 'data')
+
+
+@contextmanager
+def temp_dir():
+    """Create Temp directory for test.
+
+     Open:
+        - Create temp directory.
+
+    Close:
+        - Delete temp directory.
+    """
+    temp = UNIT_TEST_DATA / 'temp'
+    try:
+        temp.mkdir(parents=True, exist_ok=True)
+        yield temp
+    finally:
+        rmtree(temp)
+
+
+class TestXSOARConfigFileUpdater:
+
+    @pytest.mark.parametrize(argnames='add_all_marketplace_packs, expected_path, expected_outputs',
+                             argvalues=[(True, 'xsoar_config.json',
+                                         {'marketplace_packs': [{'id': 'test1', 'version': '1.0.0'}]}),
+                                        (False, '', {})])
+    def test_add_all_marketplace_packs(self, mocker, add_all_marketplace_packs, expected_path, expected_outputs):
+        """
+        Given:
+            - add_all_marketplace_packs arg as True or False
+        When:
+            - run the update_xsoar_config_file command
+        Then:
+            - validate the xsoar_config file exist in the destination output
+            - validate the xsoar_config file output is as expected
+        """
+
+        mocker.patch.object(XSOARConfigFileUpdater, 'get_installed_packs', return_value=[
+            {"id": "test1", "version": "1.0.0"}])
+        with temp_dir() as tmp_output_dir:
+            click.Context(command=xsoar_config_file_update).invoke(
+                xsoar_config_file_update,
+                file_path=tmp_output_dir / 'xsoar_config.json',
+                add_all_marketplace_packs=add_all_marketplace_packs
+            )
+
+            assert Path(f'{tmp_output_dir}/{expected_path}').exists()
+
+            try:
+                with open(f'{tmp_output_dir}/{expected_path}', 'r') as config_file:
+                    config_file_info = json.load(config_file)
+            except IsADirectoryError:
+                config_file_info = {}
+            assert config_file_info == expected_outputs
+
+    @pytest.mark.parametrize(argnames='add_marketplace_pack, pack_id, pack_data, expected_path, err, expected_outputs',
+                             argvalues=[(True, 'Pack1', '1.0.1', 'xsoar_config.json', '',
+                                         {'marketplace_packs': [{'id': 'Pack1', 'version': '1.0.1'}]})])
+    def test_add_marketplace_pack(self, add_marketplace_pack, pack_id, pack_data, expected_path, capsys, err,
+                                  expected_outputs):
+        """
+        Given:
+            - add_marketplace_pack arg as True
+        When:
+            - run the update_xsoar_config_file command
+        Then:
+            - validate the xsoar_config file exist in the destination output
+            - validate the xsoar_config file output is as expected
```

Comment:
This `Then` is a bit too verbose. Makes it hard to understand what is the actual behavior you're trying to test here.
Also, the test itself (and other tests) can be made more readable

Reply:
?

Also, using parametrize for one input is a bit hacky IMO